compiler/rustc_middle/src/ty/context.rs

```diff
@@ -1703,6 +1703,41 @@ impl<'tcx> TyCtxt<'tcx> {
         v.0
     }
 
+    /// Given a `DefId` for an `fn`, return all the `dyn` and `impl` traits in its return type and associated alias span when type alias is used
+    pub fn return_type_impl_or_dyn_traits_with_type_alias(
+        self,
+        scope_def_id: LocalDefId,
+    ) -> Option<(Vec<&'tcx hir::Ty<'tcx>>, Span)> {
+        let hir_id = self.hir().local_def_id_to_hir_id(scope_def_id);
+        let Some(hir::FnDecl { output: hir::FnRetTy::Return(hir_output), .. }) = self.hir().fn_decl_by_hir_id(hir_id) else {
+            return None;
+        };
+
+        let mut v = TraitObjectVisitor(vec![], self.hir());
+
+        // when the return type is a type alias
+        if let hir::TyKind::Path(hir::QPath::Resolved(
```

Comment:
Can you use let else here?

Reply:
Modified a bit, but not sure how to use `let else` here.